packages/cli/generation/ir-generator/src/resolvers/SourceResolver.ts

```diff
@@ -10,14 +11,19 @@ export interface SourceResolver {
         source: RawSchemas.SourceSchema;
         file: FernFileContext;
     }) => Promise<ResolvedSource | undefined>;
-    resolveSourceOrThrow: (args: { source: RawSchemas.SourceSchema; file: FernFileContext }) => Promise<ResolvedSource>;
+    resolveSourceOrThrow: (args: {
+        source: RawSchemas.SourceSchema;
+        file: FernFileContext;
+    }) => Promise<ResolvedSource | undefined>;
 }
 
 export class SourceResolverImpl implements SourceResolver {
+    private readonly taskContext: TaskContext;
```

Comment:
nit: Can we name this property `context` for consistency with elsewhere? That way we just say:

```typescript
this.context.logger.warn(`Cannot resolve source ${source.openapi} from file ${file.relativeFilepath}`);
```